src/dialects/oracle/data-types.js

```diff
@@ -0,0 +1,470 @@
+// Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved
+
+'use strict';
+
+const moment = require('moment');
+const momentTz = require('moment-timezone');
+
+module.exports = BaseTypes => {
+  const warn = BaseTypes.ABSTRACT.warn.bind(
+    undefined,
+    'https://docs.oracle.com/database/122/SQLRF/Data-Types.htm#SQLRF30020'
+  );
+
+  BaseTypes.DATE.types.oracle = ['TIMESTAMP', 'TIMESTAMP WITH LOCAL TIME ZONE'];
+  BaseTypes.STRING.types.oracle = ['VARCHAR2', 'NVARCHAR2'];
+  BaseTypes.CHAR.types.oracle = ['CHAR', 'RAW'];
+  BaseTypes.TEXT.types.oracle = ['CLOB'];
+  BaseTypes.TINYINT.types.oracle = ['NUMBER'];
+  BaseTypes.SMALLINT.types.oracle = ['NUMBER'];
+  BaseTypes.MEDIUMINT.types.oracle = ['NUMBER'];
+  BaseTypes.INTEGER.types.oracle = ['INTEGER'];
+  BaseTypes.BIGINT.types.oracle = ['NUMBER'];
+  BaseTypes.FLOAT.types.oracle = ['BINARY_FLOAT'];
+  BaseTypes.DATEONLY.types.oracle = ['DATE'];
+  BaseTypes.BOOLEAN.types.oracle = ['CHAR(1)'];
+  BaseTypes.BLOB.types.oracle = ['BLOB'];
+  BaseTypes.DECIMAL.types.oracle = ['NUMBER'];
+  BaseTypes.UUID.types.oracle = ['VARCHAR2'];
+  BaseTypes.ENUM.types.oracle = ['VARCHAR2'];
+  BaseTypes.REAL.types.oracle = ['BINARY_DOUBLE'];
+  BaseTypes.DOUBLE.types.oracle = ['BINARY_DOUBLE'];
+  BaseTypes.JSON.types.oracle = ['BLOB'];
+  BaseTypes.GEOMETRY.types.oracle = false;
+
+  class STRING extends BaseTypes.STRING {
+    toSql() {
+      if (this.length > 4000 || this._binary && this._length > 2000) {
+        warn(
+          'Oracle 12 supports length up to 32764; be sure that your administrator has extended the MAX_STRING_SIZE parameter. Check https://docs.oracle.com/database/121/REFRN/GUID-D424D23B-0933-425F-BC69-9C0E6724693C.htm#REFRN10321'
```

Comment:
Always use redirecting URLs to get the latest doc: https://www.oracle.com/pls/topic/lookup?ctx=dblatest&id=GUID-D424D23B-0933-425F-BC69-9C0E6724693C

I commented on two Oracle doc URLs.  There may be others in the PR that need updating

Reply:
Done